signer/src/network/libp2p/event_loop.rs

```diff
@@ -338,11 +338,25 @@ fn handle_gossipsub_event(
             message,
             ..
         } => {
-            if !ctx.state().current_signer_set().is_allowed_peer(&peer_id) {
+            let current_signer_set = ctx.state().current_signer_set();
+            if !current_signer_set.is_allowed_peer(&peer_id) {
```

Comment:
note: this check is technically unnecessary since we can't receive a message via a peer if it's not connected, and you can't connect if not a valid peer. The origin check below was a good catch, though!

Reply:
Oh right. I'll add a comment that it is unnecessary.